test/test_torch.py

```diff
@@ -5993,6 +5993,37 @@ def test_diagonal_multidim(self, device, dtype):
         self.assertEqual(expected.shape, result.shape)
         self.assertEqual(expected, result)
 
+    def _test_trace(self, device, dtype, legacy):
+        def test(shape):
+            tensor = make_tensor(shape, device, dtype, low=-9, high=9)
+            diag = tensor.diag()
+            if legacy:
+                # NB: trace on cpu doesn't do type promotion...
```

Comment:
good catch! once the issue is filed, can you also link it here in the comment?

Reply:
https://github.com/pytorch/pytorch/issues/47127

Reply:
tossed into comment